src/main/java/com/opentable/metrics/graphite/GraphiteSenderWrapper.java

```diff
@@ -117,19 +121,16 @@ public synchronized int getFailures() {
     }
 
     private synchronized Graphite delegate() throws IOException {
-        final boolean needsReconnectFail = needsReconnectFail(delegate);
-        final boolean needsReconnectPeriodic = needsReconnectPeriodic();
-        if (needsReconnectFail || needsReconnectPeriodic) {
+        final boolean explicitlyClosed = delegate == null;
+        boolean needsReconnectFail = false;
+        if (explicitlyClosed || (needsReconnectFail = needsReconnectFail(delegate)) || needsReconnectPeriodic()) {
             if (needsReconnectFail) {
                 connectionFailures.inc();
-                long failCount = connectionFailures.getCount();
-                if (failCount > 0) {
-                    LOG.warn("bad graphite state; recycling; connected {}, failures {}; counter {}; last @ {}",
-                            delegate == null ? "UNKNOWN" : delegate.isConnected(),
-                            delegate == null ? "UNKNOWN" : delegate.getFailures(),
-                            failCount,
-                            lastReconnect);
-                }
+                LOG.warn("bad graphite state; recycling; connected {}, failures {}; counter {}; last @ {}",
+                        delegate == null ? "UNKNOWN" : delegate.isConnected(),
+                        delegate == null ? "UNKNOWN" : delegate.getFailures(),
```

Comment:
is there a reason for not using `explicitlyClosed` here.

Reply:
Oh, that's not a bad idea.  Although the naming might be a bit confusing.  It happens to be the case that the delegate is closed iff the reference is `null`.  But if you were to read something like `explicitlyClosed ? "UNKNOWN" : delegate.isConnected()` here, would that really be clear?  Seems more like the `null` check ternaries on these lines are more about an explicit `null` check, as opposed to logic related to whether the connection is "closed".  What do you think?